scripts/generate_enums.py

```diff
@@ -0,0 +1,356 @@
+                for line in lines:
+                    doc_string += line + "\n"
+
+        return doc_string
+
+    def doc_item_to_string(self, doc_item):
+        result = []
+        for item in doc_item:
+            t = item['type']
+            if t == "text":
+                result.append(item['text'])
+            elif t == "ref" or t == "code":
+                result.append(self.ref_to_string(item['text']))
+            elif t == "list":
+                ul = "\n"
+                for li in item['values']:
+                    ul += f" - {self.doc_item_to_string(li)}\n"
+                ul += "\n"
+                result.append(ul)
+            else:
+                print(f"unknown doc item type = {t}, consider adding it to doc_item_to_string")
+        return ' '.join(result).replace(" ,", ",").replace(" .", ".")
+
+    def ref_to_string(self, ref_value):
+        """
+        :param ref_value: enum_value or enum_type, e.g. roc_endpoint or ROC_INTERFACE_CONSOLIDATED
+        :return: go value
+        """
+        if ref_value.startswith("roc_"):
+            return to_camel_case(ref_value.removeprefix('roc_'))
+
+        return to_camel_case(ref_value.lower().removeprefix('roc_'))
+
+
+def parse_config_xml():
+    try:
+        tree = ElementTree.parse(CONFIG_FILE_PATH)
+        return tree.getroot()
+    except FileNotFoundError:
+        print(f"File not found: {CONFIG_FILE_PATH}")
+        exit(1)
+    except ElementTree.ParseError:
+        print(f"Error parsing XML file: {CONFIG_FILE_PATH}")
+        exit(1)
+
+
+def get_enums(root):
+    enum_definitions = []
+    enum_memberdefs = root.findall('.//sectiondef[@kind="enum"]/memberdef[@kind="enum"]')
+    for member_def in enum_memberdefs:
+        name = member_def.find('name').text
+        print(f"found enum in docs: {name}")
+        doc = parse_doc(member_def)
+        values = []
+
+        for enum_value in member_def.findall('enumvalue'):
+            enum_name = enum_value.find('name').text
+            value = enum_value.find('initializer').text.removeprefix('= ')
+            enum_value_doc = parse_doc(enum_value)
+            values.append(EnumValue(enum_name, value, enum_value_doc))
+
+        enum_definitions.append(EnumDefinition(name, values, doc))
+
+    return enum_definitions
+
+
+def parse_doc(elem):
+    items = []
+    brief = elem.find('briefdescription/para')
+    items.append(parse_doc_elem(brief))
+
+    for para in elem.findall('detaileddescription/para'):
+        items.append(parse_doc_elem(para))
+
+    return DocComment(items)
+
+
+def strip_text(text):
+    if text:
+        strip = text.strip()
+        if strip:
+            return strip
+    return None
+
+
+def parse_doc_elem(elem: ElementTree.Element):
+    items = []
+    tag = elem.tag
+    parse_children = True
+    text = strip_text(elem.text)
+    if tag == "para":
+        if text:
+            items.append({"type": "text", "text": text})
```

Comment:
nit: consider using namedtuple or dataclass for items

Reply:
done